### PR TITLE
tidb: add `general-log-file` and `compression`

### PR DESCRIPTION
**Pull request by hfxsd**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/16719
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by Defined2014 — APPROVED** [review]

File: tidb-configuration-file.md [hunk 1]
```diff
@@ -319,6 +319,12 @@ Configuration items related to log.
 - Default value: `10000`
 - When the number of query rows (including the intermediate results based on statistics) is larger than this value, it is an `expensive` operation and outputs log with the `[EXPENSIVE_QUERY]` prefix.
 
+### `general-log-file` <span class="version-mark">New in v8.0.0</span>
+
++ The file name of the [general log](/system-variables.md#tidb_general_log).
++ Default value: ""
++ If you set file name, the general log is output to this file. If the value is empty, the general log is output to the file specified by [`filename`](#filename) on the TiDB server.
```
Comment (hfxsd) [inline tidb-configuration-file.md:326]:
```suggestion
+ If you specify a file name, the general log is output to this file. If the value is empty, the general log is output to the file specified by [`filename`](#filename) on the TiDB server.
```

File: tidb-configuration-file.md [hunk 2]
```diff
@@ -319,6 +319,12 @@ Configuration items related to log.
 - Default value: `10000`
 - When the number of query rows (including the intermediate results based on statistics) is larger than this value, it is an `expensive` operation and outputs log with the `[EXPENSIVE_QUERY]` prefix.
 
+### `general-log-file` <span class="version-mark">New in v8.0.0</span>
+
++ The file name of the [general log](/system-variables.md#tidb_general_log).
++ Default value: ""
++ If you specify a file name, the general log is output to this file. If the value is empty, the general log is output to the file specified by [`filename`](#filename) on the TiDB server.
```
Comment (songrijie) [inline tidb-configuration-file.md:326]:
```suggestion
+ If you specify a file name, the general log is written to this particular file. If the value is blank, the general log is written to the server log of the TiDB instance. The name of server log is specified by [`filename`](#filename) . 
```

**Review by songrijie — APPROVED** [review]

**Review by bb7133 — APPROVED** [review]
LGTM

File: tidb-configuration-file.md [hunk 3]
```diff
@@ -319,6 +319,12 @@ Configuration items related to log.
 - Default value: `10000`
 - When the number of query rows (including the intermediate results based on statistics) is larger than this value, it is an `expensive` operation and outputs log with the `[EXPENSIVE_QUERY]` prefix.
 
+### `general-log-file` <span class="version-mark">New in v8.0.0</span>
+
++ The file name of the [general log](/system-variables.md#tidb_general_log).
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:324]:
```suggestion
+ The filename of the [general log](/system-variables.md#tidb_general_log).
```
https://developers.google.com/style/word-list#filename

File: tidb-configuration-file.md [hunk 4]
```diff
@@ -319,6 +319,12 @@ Configuration items related to log.
...
++ Default value: ""
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:325]:
```suggestion
+ Default value: `""`
```

File: tidb-configuration-file.md [hunk 5]
```diff
@@ -319,6 +319,12 @@ Configuration items related to log.
...
++ If you specify a file name, the general log is written to this particular file. If the value is blank, the general log is written to the server log of the TiDB instance. The name of server log is specified by [`filename`](#filename).
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:326]:
```suggestion
+ If you specify a filename, the general log is written to this specified file. If the value is blank, the general log is written to the server log of the TiDB instance. You can specify the name of the server log using [`filename`](#filename).
```

File: tidb-configuration-file.md [hunk 6]
```diff
@@ -355,6 +361,12 @@ Configuration items related to log files.
 - Default value: `0`
 - All the log files are retained by default. If you set it to `7`, seven log files are retained at maximum.
 
+#### `compress` <span class="version-mark">New in v8.0.0</span>
+
++ The compression method for the log.
++ Default value: ""
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:367]:
```suggestion
+ Default value: `""`
+ Value options: `""`, `"gzip"`
```

File: tidb-configuration-file.md [hunk 7]
```diff
@@ -355,6 +361,12 @@ Configuration items related to log files.
...
++ By default, compression is not enabled. Currently, the only supported compression format is `"gzip"`. After compression is enabled, all log files are affected, such as `slow-log-file` and `general-log-file`.
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:368]:
```suggestion
+ The default value is `""`, which means no compression. To enable the gzip compression, set this value to `"gzip"`. After compression is enabled, all log files are affected, such as [`slow-query-file`](#slow-query-file) and [`general-log-file`](#general-log-file).
```

File: tidb-configuration-file.md [hunk 8]
```diff
@@ -355,6 +361,12 @@ Configuration items related to log files.
 - Default value: `0`
 - All the log files are retained by default. If you set it to `7`, seven log files are retained at maximum.
 
+#### `compress` <span class="version-mark">New in v8.0.0</span>
```
Comment (Defined2014) [inline tidb-configuration-file.md:364]:
```suggestion
#### `compression` <span class="version-mark">New in v8.0.0</span>
```

File: command-line-flags-for-tidb-configuration.md [hunk 9]
```diff
@@ -79,6 +79,12 @@ When you start the TiDB cluster, you can use command-line options or environment
 - Default: `""`
 - If this option is not set, logs are output to "stderr". If this option is set, logs are output to the corresponding file.
 
+## `--log-general`
+
++ The filename of the [General Log](/system-variables.md#tidb_general_log)
++ Default: `""`
++ If this option is not set, the general log is output to the file specified by `--log-file` by default.
```
Comment (Oreoxmt) [inline command-line-flags-for-tidb-configuration.md:86]:
```suggestion
+ If this option is not set, the general log is written to the file specified by [`--log-file`](#--log-file) by default.
```

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**